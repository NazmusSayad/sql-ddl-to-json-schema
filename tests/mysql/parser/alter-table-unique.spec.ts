import { getParsedFormat } from '../parse-handler';
import { run } from '../runner';

run(getParsedFormat, {
  'Parser: Should alter table adding unique key with index option, two columns and options.': {
    queries: [
      `ALTER TABLE people add constraint xyz unique key ik_id using hash ( id ( 2 ) asc , o_id ) key_block_size 1024 comment 'test';`,
    ],
  },

  'Parser: Should alter table adding unique key with index option, two columns and option.': {
    queries: [
      `ALTER TABLE people add constraint unique key ik_id using hash (id(2) asc, o_id ) comment 'test';`,
      `ALTER TABLE people add unique key ik_id using hash (id(2) asc, o_id ) comment 'test';`,
    ],
  },

  'Parser: Should alter table adding unique key with one column and option.': {
    queries: [`ALTER TABLE people add unique key ik_id(id)key_block_size 1024;`],
  },

  'Parser: Should alter table adding unique key with one column.': {
    queries: [
      `ALTER TABLE people add unique key ik_id (id);`,
      `ALTER TABLE people add unique key ik_id(id);`,
    ],
  },

  'Parser: Should alter table adding unnamed unique key with one column.': {
    queries: [`ALTER TABLE people add unique key (id);`, `ALTER TABLE people add unique key(id);`],
  },

  'Parser: Should alter table adding unique index with index option, two columns and options.': {
    queries: [
      `ALTER TABLE people add constraint xyz unique index ik_id using hash ( id ( 2 ) asc , o_id ) key_block_size 1024 comment 'test';`,
    ],
  },

  'Parser: Should alter table adding unique index with index option, two columns and option.': {
    queries: [
      `ALTER TABLE people add constraint unique index ik_id using hash ( id(2) asc, o_id ) comment 'test';`,
      `ALTER TABLE people add unique index ik_id using hash ( id(2) asc, o_id ) comment 'test';`,
    ],
  },

  'Parser: Should alter table adding unique index with one column and option.': {
    queries: [`ALTER TABLE people add unique index ik_id(id)key_block_size 1024;`],
  },

  'Parser: Should alter table adding unique index with one column.': {
    queries: [
      `ALTER TABLE people add unique index ik_id (id);`,
      `ALTER TABLE people add unique index ik_id(id);`,
    ],
  },

  'Parser: Should alter table adding unnamed unique index with one column.': {
    queries: [
      `ALTER TABLE people add unique index (id);`,
      `ALTER TABLE people add unique index(id);`,
    ],
  },
});
