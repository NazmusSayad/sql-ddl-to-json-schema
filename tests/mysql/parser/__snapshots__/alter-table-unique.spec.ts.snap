// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Parser: Should alter table adding unique index with index option, two columns and option. 1`] = `
{
  "id": "MAIN",
  "def": [
    {
      "id": "P_DDS",
      "def": {
        "id": "P_ALTER_TABLE",
        "def": {
          "table": "people",
          "specs": [
            {
              "id": "P_ALTER_TABLE_SPECS",
              "def": {
                "spec": {
                  "id": "O_ALTER_TABLE_SPEC",
                  "def": {
                    "action": "addUniqueKey",
                    "name": null,
                    "index": {
                      "id": "P_INDEX_TYPE",
                      "def": "hash"
                    },
                    "columns": [
                      {
                        "id": "P_INDEX_COLUMN",
                        "def": {
                          "column": "id",
                          "length": 2,
                          "sort": "asc"
                        }
                      },
                      {
                        "id": "P_INDEX_COLUMN",
                        "def": {
                          "column": "o_id"
                        }
                      }
                    ],
                    "options": [
                      {
                        "id": "O_INDEX_OPTION",
                        "def": {
                          "comment": "test"
                        }
                      }
                    ]
                  }
                }
              }
            }
          ]
        }
      }
    }
  ]
}
`;

exports[`Parser: Should alter table adding unique index with index option, two columns and option. 2`] = `
{
  "id": "MAIN",
  "def": [
    {
      "id": "P_DDS",
      "def": {
        "id": "P_ALTER_TABLE",
        "def": {
          "table": "people",
          "specs": [
            {
              "id": "P_ALTER_TABLE_SPECS",
              "def": {
                "spec": {
                  "id": "O_ALTER_TABLE_SPEC",
                  "def": {
                    "action": "addUniqueKey",
                    "name": null,
                    "index": {
                      "id": "P_INDEX_TYPE",
                      "def": "hash"
                    },
                    "columns": [
                      {
                        "id": "P_INDEX_COLUMN",
                        "def": {
                          "column": "id",
                          "length": 2,
                          "sort": "asc"
                        }
                      },
                      {
                        "id": "P_INDEX_COLUMN",
                        "def": {
                          "column": "o_id"
                        }
                      }
                    ],
                    "options": [
                      {
                        "id": "O_INDEX_OPTION",
                        "def": {
                          "comment": "test"
                        }
                      }
                    ]
                  }
                }
              }
            }
          ]
        }
      }
    }
  ]
}
`;

exports[`Parser: Should alter table adding unique index with index option, two columns and options. 1`] = `
{
  "id": "MAIN",
  "def": [
    {
      "id": "P_DDS",
      "def": {
        "id": "P_ALTER_TABLE",
        "def": {
          "table": "people",
          "specs": [
            {
              "id": "P_ALTER_TABLE_SPECS",
              "def": {
                "spec": {
                  "id": "O_ALTER_TABLE_SPEC",
                  "def": {
                    "action": "addUniqueKey",
                    "name": "xyz",
                    "index": {
                      "id": "P_INDEX_TYPE",
                      "def": "hash"
                    },
                    "columns": [
                      {
                        "id": "P_INDEX_COLUMN",
                        "def": {
                          "column": "id",
                          "length": 2,
                          "sort": "asc"
                        }
                      },
                      {
                        "id": "P_INDEX_COLUMN",
                        "def": {
                          "column": "o_id"
                        }
                      }
                    ],
                    "options": [
                      {
                        "id": "O_INDEX_OPTION",
                        "def": {
                          "keyBlockSize": 1024
                        }
                      },
                      {
                        "id": "O_INDEX_OPTION",
                        "def": {
                          "comment": "test"
                        }
                      }
                    ]
                  }
                }
              }
            }
          ]
        }
      }
    }
  ]
}
`;

exports[`Parser: Should alter table adding unique index with one column and option. 1`] = `
{
  "id": "MAIN",
  "def": [
    {
      "id": "P_DDS",
      "def": {
        "id": "P_ALTER_TABLE",
        "def": {
          "table": "people",
          "specs": [
            {
              "id": "P_ALTER_TABLE_SPECS",
              "def": {
                "spec": {
                  "id": "O_ALTER_TABLE_SPEC",
                  "def": {
                    "action": "addUniqueKey",
                    "name": null,
                    "index": null,
                    "columns": [
                      {
                        "id": "P_INDEX_COLUMN",
                        "def": {
                          "column": "id"
                        }
                      }
                    ],
                    "options": [
                      {
                        "id": "O_INDEX_OPTION",
                        "def": {
                          "keyBlockSize": 1024
                        }
                      }
                    ]
                  }
                }
              }
            }
          ]
        }
      }
    }
  ]
}
`;

exports[`Parser: Should alter table adding unique index with one column. 1`] = `
{
  "id": "MAIN",
  "def": [
    {
      "id": "P_DDS",
      "def": {
        "id": "P_ALTER_TABLE",
        "def": {
          "table": "people",
          "specs": [
            {
              "id": "P_ALTER_TABLE_SPECS",
              "def": {
                "spec": {
                  "id": "O_ALTER_TABLE_SPEC",
                  "def": {
                    "action": "addUniqueKey",
                    "name": null,
                    "index": null,
                    "columns": [
                      {
                        "id": "P_INDEX_COLUMN",
                        "def": {
                          "column": "id"
                        }
                      }
                    ],
                    "options": []
                  }
                }
              }
            }
          ]
        }
      }
    }
  ]
}
`;

exports[`Parser: Should alter table adding unique index with one column. 2`] = `
{
  "id": "MAIN",
  "def": [
    {
      "id": "P_DDS",
      "def": {
        "id": "P_ALTER_TABLE",
        "def": {
          "table": "people",
          "specs": [
            {
              "id": "P_ALTER_TABLE_SPECS",
              "def": {
                "spec": {
                  "id": "O_ALTER_TABLE_SPEC",
                  "def": {
                    "action": "addUniqueKey",
                    "name": null,
                    "index": null,
                    "columns": [
                      {
                        "id": "P_INDEX_COLUMN",
                        "def": {
                          "column": "id"
                        }
                      }
                    ],
                    "options": []
                  }
                }
              }
            }
          ]
        }
      }
    }
  ]
}
`;

exports[`Parser: Should alter table adding unique key with index option, two columns and option. 1`] = `
{
  "id": "MAIN",
  "def": [
    {
      "id": "P_DDS",
      "def": {
        "id": "P_ALTER_TABLE",
        "def": {
          "table": "people",
          "specs": [
            {
              "id": "P_ALTER_TABLE_SPECS",
              "def": {
                "spec": {
                  "id": "O_ALTER_TABLE_SPEC",
                  "def": {
                    "action": "addUniqueKey",
                    "name": null,
                    "index": {
                      "id": "P_INDEX_TYPE",
                      "def": "hash"
                    },
                    "columns": [
                      {
                        "id": "P_INDEX_COLUMN",
                        "def": {
                          "column": "id",
                          "length": 2,
                          "sort": "asc"
                        }
                      },
                      {
                        "id": "P_INDEX_COLUMN",
                        "def": {
                          "column": "o_id"
                        }
                      }
                    ],
                    "options": [
                      {
                        "id": "O_INDEX_OPTION",
                        "def": {
                          "comment": "test"
                        }
                      }
                    ]
                  }
                }
              }
            }
          ]
        }
      }
    }
  ]
}
`;

exports[`Parser: Should alter table adding unique key with index option, two columns and option. 2`] = `
{
  "id": "MAIN",
  "def": [
    {
      "id": "P_DDS",
      "def": {
        "id": "P_ALTER_TABLE",
        "def": {
          "table": "people",
          "specs": [
            {
              "id": "P_ALTER_TABLE_SPECS",
              "def": {
                "spec": {
                  "id": "O_ALTER_TABLE_SPEC",
                  "def": {
                    "action": "addUniqueKey",
                    "name": null,
                    "index": {
                      "id": "P_INDEX_TYPE",
                      "def": "hash"
                    },
                    "columns": [
                      {
                        "id": "P_INDEX_COLUMN",
                        "def": {
                          "column": "id",
                          "length": 2,
                          "sort": "asc"
                        }
                      },
                      {
                        "id": "P_INDEX_COLUMN",
                        "def": {
                          "column": "o_id"
                        }
                      }
                    ],
                    "options": [
                      {
                        "id": "O_INDEX_OPTION",
                        "def": {
                          "comment": "test"
                        }
                      }
                    ]
                  }
                }
              }
            }
          ]
        }
      }
    }
  ]
}
`;

exports[`Parser: Should alter table adding unique key with index option, two columns and options. 1`] = `
{
  "id": "MAIN",
  "def": [
    {
      "id": "P_DDS",
      "def": {
        "id": "P_ALTER_TABLE",
        "def": {
          "table": "people",
          "specs": [
            {
              "id": "P_ALTER_TABLE_SPECS",
              "def": {
                "spec": {
                  "id": "O_ALTER_TABLE_SPEC",
                  "def": {
                    "action": "addUniqueKey",
                    "name": "xyz",
                    "index": {
                      "id": "P_INDEX_TYPE",
                      "def": "hash"
                    },
                    "columns": [
                      {
                        "id": "P_INDEX_COLUMN",
                        "def": {
                          "column": "id",
                          "length": 2,
                          "sort": "asc"
                        }
                      },
                      {
                        "id": "P_INDEX_COLUMN",
                        "def": {
                          "column": "o_id"
                        }
                      }
                    ],
                    "options": [
                      {
                        "id": "O_INDEX_OPTION",
                        "def": {
                          "keyBlockSize": 1024
                        }
                      },
                      {
                        "id": "O_INDEX_OPTION",
                        "def": {
                          "comment": "test"
                        }
                      }
                    ]
                  }
                }
              }
            }
          ]
        }
      }
    }
  ]
}
`;

exports[`Parser: Should alter table adding unique key with one column and option. 1`] = `
{
  "id": "MAIN",
  "def": [
    {
      "id": "P_DDS",
      "def": {
        "id": "P_ALTER_TABLE",
        "def": {
          "table": "people",
          "specs": [
            {
              "id": "P_ALTER_TABLE_SPECS",
              "def": {
                "spec": {
                  "id": "O_ALTER_TABLE_SPEC",
                  "def": {
                    "action": "addUniqueKey",
                    "name": null,
                    "index": null,
                    "columns": [
                      {
                        "id": "P_INDEX_COLUMN",
                        "def": {
                          "column": "id"
                        }
                      }
                    ],
                    "options": [
                      {
                        "id": "O_INDEX_OPTION",
                        "def": {
                          "keyBlockSize": 1024
                        }
                      }
                    ]
                  }
                }
              }
            }
          ]
        }
      }
    }
  ]
}
`;

exports[`Parser: Should alter table adding unique key with one column. 1`] = `
{
  "id": "MAIN",
  "def": [
    {
      "id": "P_DDS",
      "def": {
        "id": "P_ALTER_TABLE",
        "def": {
          "table": "people",
          "specs": [
            {
              "id": "P_ALTER_TABLE_SPECS",
              "def": {
                "spec": {
                  "id": "O_ALTER_TABLE_SPEC",
                  "def": {
                    "action": "addUniqueKey",
                    "name": null,
                    "index": null,
                    "columns": [
                      {
                        "id": "P_INDEX_COLUMN",
                        "def": {
                          "column": "id"
                        }
                      }
                    ],
                    "options": []
                  }
                }
              }
            }
          ]
        }
      }
    }
  ]
}
`;

exports[`Parser: Should alter table adding unique key with one column. 2`] = `
{
  "id": "MAIN",
  "def": [
    {
      "id": "P_DDS",
      "def": {
        "id": "P_ALTER_TABLE",
        "def": {
          "table": "people",
          "specs": [
            {
              "id": "P_ALTER_TABLE_SPECS",
              "def": {
                "spec": {
                  "id": "O_ALTER_TABLE_SPEC",
                  "def": {
                    "action": "addUniqueKey",
                    "name": null,
                    "index": null,
                    "columns": [
                      {
                        "id": "P_INDEX_COLUMN",
                        "def": {
                          "column": "id"
                        }
                      }
                    ],
                    "options": []
                  }
                }
              }
            }
          ]
        }
      }
    }
  ]
}
`;

exports[`Parser: Should alter table adding unnamed unique index with one column. 1`] = `
{
  "id": "MAIN",
  "def": [
    {
      "id": "P_DDS",
      "def": {
        "id": "P_ALTER_TABLE",
        "def": {
          "table": "people",
          "specs": [
            {
              "id": "P_ALTER_TABLE_SPECS",
              "def": {
                "spec": {
                  "id": "O_ALTER_TABLE_SPEC",
                  "def": {
                    "action": "addUniqueKey",
                    "name": null,
                    "index": null,
                    "columns": [
                      {
                        "id": "P_INDEX_COLUMN",
                        "def": {
                          "column": "id"
                        }
                      }
                    ],
                    "options": []
                  }
                }
              }
            }
          ]
        }
      }
    }
  ]
}
`;

exports[`Parser: Should alter table adding unnamed unique index with one column. 2`] = `
{
  "id": "MAIN",
  "def": [
    {
      "id": "P_DDS",
      "def": {
        "id": "P_ALTER_TABLE",
        "def": {
          "table": "people",
          "specs": [
            {
              "id": "P_ALTER_TABLE_SPECS",
              "def": {
                "spec": {
                  "id": "O_ALTER_TABLE_SPEC",
                  "def": {
                    "action": "addUniqueKey",
                    "name": null,
                    "index": null,
                    "columns": [
                      {
                        "id": "P_INDEX_COLUMN",
                        "def": {
                          "column": "id"
                        }
                      }
                    ],
                    "options": []
                  }
                }
              }
            }
          ]
        }
      }
    }
  ]
}
`;

exports[`Parser: Should alter table adding unnamed unique key with one column. 1`] = `
{
  "id": "MAIN",
  "def": [
    {
      "id": "P_DDS",
      "def": {
        "id": "P_ALTER_TABLE",
        "def": {
          "table": "people",
          "specs": [
            {
              "id": "P_ALTER_TABLE_SPECS",
              "def": {
                "spec": {
                  "id": "O_ALTER_TABLE_SPEC",
                  "def": {
                    "action": "addUniqueKey",
                    "name": null,
                    "index": null,
                    "columns": [
                      {
                        "id": "P_INDEX_COLUMN",
                        "def": {
                          "column": "id"
                        }
                      }
                    ],
                    "options": []
                  }
                }
              }
            }
          ]
        }
      }
    }
  ]
}
`;

exports[`Parser: Should alter table adding unnamed unique key with one column. 2`] = `
{
  "id": "MAIN",
  "def": [
    {
      "id": "P_DDS",
      "def": {
        "id": "P_ALTER_TABLE",
        "def": {
          "table": "people",
          "specs": [
            {
              "id": "P_ALTER_TABLE_SPECS",
              "def": {
                "spec": {
                  "id": "O_ALTER_TABLE_SPEC",
                  "def": {
                    "action": "addUniqueKey",
                    "name": null,
                    "index": null,
                    "columns": [
                      {
                        "id": "P_INDEX_COLUMN",
                        "def": {
                          "column": "id"
                        }
                      }
                    ],
                    "options": []
                  }
                }
              }
            }
          ]
        }
      }
    }
  ]
}
`;
